addon/mixins/in-viewport.js

```diff
@@ -3,53 +3,62 @@
 
 import Ember from 'ember';
 
-var on           = Ember.on;
-var get          = Ember.get;
-var Mixin        = Ember.Mixin;
-var debounce     = Ember.run.debounce;
+var on = Ember.on;
+var get = Ember.get;
+var Mixin = Ember.Mixin;
+var debounce = Ember.run.debounce;
 var scheduleOnce = Ember.run.scheduleOnce;
 
 export default Mixin.create({
-  scrollTimeout:   100,
+  scrollTimeout: 100,
   enteredViewport: null,
+  threshold: 0,
 
-  _setViewport: function() {
+  _setViewport: function () {
     var rect = this.$()[0].getBoundingClientRect();
 
-    this.set('enteredViewport',
-      rect.top >= 0 &&
+    var enteredViewport = rect.top >= 0 &&
       rect.left >= 0 &&
-      rect.bottom <= (window.innerHeight || document.documentElement.clientHeight) &&
-      rect.right <= (window.innerWidth || document.documentElement.clientWidth)
-    );
+      (rect.bottom - get(this, 'threshold')) <= (window.innerHeight || document.documentElement.clientHeight) &&
+      rect.right <= (window.innerWidth || document.documentElement.clientWidth);
+
+    this.set('enteredViewport', enteredViewport);
+
+    if (enteredViewport) {
+      this._unbindScroll();
```

Comment:
This part, by the way, is performance improvement. We unbind scroll event when image is already loaded, as if you have 40 images on the pages, each image continue to listen for the scroll event.